tests/MongoDB.Driver.Examples/ChangeStreamExamples.cs

```diff
@@ -122,5 +126,77 @@ public void ChangeStreamExample3()
                 next.FullDocument.Should().Be(documents[1]);
             }
         }
+
+        [Fact]
+        public void Changestream_Example_4()
+        {
+            RequireServer.Check().Supports(Feature.AggregateAddFields);
+
+            var client = DriverTestConfiguration.Client;
+            var database = client.GetDatabase("ChangeStreamExamples");
+            database.DropCollection("inventory");
+            var inventoryCollection = database.GetCollection<BsonDocument>("inventory");
+
+            var cancelationTokenSource = new CancellationTokenSource();
+            try
+            {
+                BsonDocument document = new BsonDocument("username", "alice");
+
+                Task.Run(() =>
+                {
+                    while (!cancelationTokenSource.IsCancellationRequested)
+                    {
+                        Thread.Sleep(TimeSpan.FromMilliseconds(100));
+                        document["_id"] = Guid.NewGuid();
+                        inventoryCollection.InsertOne(document);
+                    }
+                }, cancelationTokenSource.Token);
+
+                // Currently no helper method for $addFields
+                // Start Changestream Example 4
+                var pipeline = new EmptyPipelineDefinition<ChangeStreamDocument<BsonDocument>>()
+                    .Match(filter =>
+                        filter.FullDocument["username"] == "alice" ||
+                        filter.OperationType == ChangeStreamOperationType.Delete)
+                    .AppendStage(
+                        new BsonDocumentPipelineStageDefinition<ChangeStreamDocument<BsonDocument>, BsonDocument>(
+                            new BsonDocument
+                            {
+                                {"$addFields", new BsonDocument("newField", "this is an added field!")}
+                            }));
+
+                var collection = database.GetCollection<BsonDocument>("inventory");
+                var changeStream = collection.Watch(pipeline);
+
+                var enumerator = changeStream.ToEnumerable().GetEnumerator();
+                enumerator.MoveNext();
+                var next = enumerator.Current;
+                // End Changestream Example 4
+
+                changeStream.Dispose();
+                enumerator.Dispose();
```

Comment:
There was no any mentioning of this in the ticket example, but I agree that it's better to add this calling to example brackets.

But I don't know about `using` operator. Because in this case, we will not be able to check operation result in an assert section.